### PR TITLE
fix(init-command): fix error handling

### PR DESCRIPTION
**Pull request by adriguy**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Create automatic tests
- [x] No automatic tests failures
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: src/commands/init.js [hunk 1]
```diff
@@ -61,9 +61,7 @@ class InitCommand extends AbstractAuthenticatedCommand {
 
   async projectSelection() {
     const parsed = this.parse(InitCommand);
-    spinner.pause();
```
Comment (adriguy) [inline src/commands/init.js:64]: I remove it from here to be able to have the spinner under error:
image
This is what I had if I leave it here. 
And moving it to the `with-current-project`:
image

File: src/utils/error.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ function handleError(rawError) {
   let error;
   // NOTICE: We check if the errors are from the API (have a status) or if thrown internally.
   if (rawError.status) {
+    if (rawError.status === 500) {
+      return UNEXPECTED_ERROR_MESSAGE;
+    }
```
Comment (adriguy) [inline src/utils/error.js:11]: I took the opportunity to be sure we'll never have this kind of message:
image

**Comment by adriguy** [comment]
@larcin there is one thing that I noticed that is not perfect:
image
The project not found is displayed under the "analyzing your setup"...
But this is because of the way the `with-current-project` is build, if a --project is given it return the id without checking if the project exists.
 I am not sure we want to change this unless you have an easy solution 🤔

File: src/services/with-current-project.js [hunk 3]
```diff
@@ -20,12 +24,15 @@ module.exports = async function withCurrentProject(config) {
       return { ...config, projectId: projects[0].id };
     }
 
+    const existingSpinner = spinner.isRunning();
+    if (existingSpinner) spinner.pause();
```
Comment (larcin) [inline src/services/with-current-project.js:28]: [MISC] You could add a comment here to specify that this is intended for usage outside of a spinner context

File: src/utils/error.js (hunk 2 above)
Comment (larcin) [reply src/utils/error.js:11]: [MISC] I'm not quite sure that we should do something like that. 500 errors shouldn't exist, that mens we should look for it directly when needed.

IMO we should have those displayed

**Pull request merged**